src/rfc3161_client/verify.py

```diff
@@ -213,12 +213,18 @@ def _verify_leaf_certs(self, tsp_response: TimeStampResponse) -> bool:
             leaf_certificate = self._tsa_certificate
 
         critical_eku = False
+        valid_eku = False
         for extension in leaf_certificate.extensions:
             # EKUOID is the Extended Key Usage OID, per RFC 5280
             if extension.oid == cryptography.x509.ObjectIdentifier("2.5.29.37"):
+                valid_eku = True
                 critical_eku = extension.critical
+        
+        if not valid_eku:
+            msg = "The certificate does not contain the Timesatmping EKU extension."
```

Comment:
warning, there is a typo here

```suggestion
            msg = "The certificate does not contain the Timestamping EKU extension."
```